### PR TITLE
Feature kh2 web 10/setup ci/cd

### PR DESCRIPTION
**Pull request by dima-diachkov**
Set up CI/CD
https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=3371&projectKey=KH2WEB&view=detail&selectedIssue=KH2WEB-10

1. added ESLint to dependencies and config eslintrc for some simple rules
2. created the main workflow as a pipeline lint -> test -> build -> deploy
    to test, build and deploy are using tools from GitHub marketplace
    - test, build (https://github.com/marketplace/actions/github-action-for-yarn)
    - deploy (https://github.com/marketplace/actions/deploy-react-app-to-github-pages)
4. added homepage with value "./" to package.json

### CODE REVIEW

**Review by sergeyman — APPROVED** [review]
Could  you please comment the purpose of the changing in several words

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
+name: Main workflow
+
+on:
+  push:
+    branches:
+      - master
```
Comment (kerrimov) [inline .github/workflows/main.yml:6]: All steps should run on all pull requests. Only deploy should be on master branch

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+   runs-on: ubuntu-latest
+   needs: eslint
+   steps:
+      - uses: actions/checkout@v2
+      - uses: borales/actions-yarn@v2.3.0
+        with:
+          cmd: install
+      - uses: borales/actions-yarn@v2.3.0
+        with:
+          cmd: test
```
Comment (kerrimov) [inline .github/workflows/main.yml:29]: why cmd was used here, not `run: yarn test`?

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+   needs: test
+   steps:
+      - uses: actions/checkout@v2
+      - uses: borales/actions-yarn@v2.3.0
+        with:
+          cmd: install
+      - uses: borales/actions-yarn@v2.3.0
+        with:
+          cmd: build
+      - uses: borales/actions-yarn@v2.3.0
```
Comment (kerrimov) [inline .github/workflows/main.yml:42]: the same here

File: package.json [hunk 4]
```diff
@@ -2,6 +2,7 @@
   "name": "what-to-watch",
   "version": "0.1.0",
   "private": true,
+  "homepage": "./",
```
Comment (kerrimov) [inline package.json:5]: why it was added?

File: .eslintrc.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
+/* eslint-env node */
```
Comment (kerrimov) [inline .eslintrc.js:1]: this file should be `.json`

File: .eslintrc.js [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+        "ecmaVersion": "latest",
+        "sourceType": "module"
+    },
+    "plugins": [
+        "react",
+        "@typescript-eslint"
+    ],
+    "rules": {
+        "semi": ["error", "always"],
+        "quotes": ["error", "double"]
```
Comment (kerrimov) [inline .eslintrc.js:27]: 1. please add an error for unused import
2. create import order
3.  add prettier

File: .eslintrc.js (hunk 5 above)
Comment (dima-diachkov) [reply .eslintrc.js:1]: done

File: .github/workflows/main.yml (hunk 1 above)
Comment (dima-diachkov) [reply .github/workflows/main.yml]: main.yml is divided into two parts for CI and CD

File: .github/workflows/main.yml (hunk 2 above)
Comment (dima-diachkov) [reply .github/workflows/main.yml]: changed to yarn

File: .github/workflows/main.yml (hunk 3 above)
Comment (dima-diachkov) [reply .github/workflows/main.yml]: changed to yarn

File: .eslintrc.json [hunk 7]
```diff
@@ -0,0 +1,42 @@
+/* eslint-env node */
```
Comment (kerrimov) [inline .eslintrc.json:1]: remove this comment

File: .github/workflows/cd.yml [hunk 8]
```diff
@@ -0,0 +1,15 @@
+name: cd workflow
+
+on:
+  push:
+    branches:
+      - master
+    
+jobs:
+ deploy:
+   name: Deploy
```
Comment (kerrimov) [inline .github/workflows/cd.yml:10]: needs: other steps

File: .eslintrc.json (hunk 7 above)
Comment (dima-diachkov) [reply .eslintrc.json:1]: removed

File: .github/workflows/main.yml (hunk 1 above)
Comment (dima-diachkov) [reply .github/workflows/main.yml]: united workflows into one with the condition whether there is a push to the master

File: .github/workflows/cd.yml (hunk 8 above)
Comment (dima-diachkov) [reply .github/workflows/cd.yml]: united workflows into one with the condition whether there is a push to the master

**Review by kerrimov — COMMENTED** [review]
please resolve conflicts

File: src/customizing/customTheme.tsx [hunk 9]
```diff
@@ -1,6 +1,5 @@
-import React from "react";
 import { createTheme } from "@mui/material";
-import variables from "../shared/variables.scss"
+import variables from "../shared/variables.scss";
 
 const customTheme = createTheme({
```
Comment (kerrimov) [inline src/customizing/customTheme.tsx:4]: rename folder customizing to theme

Comment (dima-diachkov) [reply src/customizing/customTheme.tsx:4]: renamed

**Review by kerrimov — APPROVED** [review]

**Pull request merged**